pypeclub/OpenPype
# Application launch stdout/stderr in GUI build

## Issue
Since in built openpype_gui executable is `sys.stdout` and `sys.stderr` set to `None` application launched using `subprocess.Popen` may get blocking handler for output which can cause crash in specific situations. It's small probability and in most of cases it will crash silently but there are cases when this cause unexpected crash of whole application (happened once to me). Most of applications do not rely on stdout set on launch, like maya, But some of them do, like nuke or non python hosts using openpype itself.

## How to replicate
- I've run a lot processes of applications and triggered threaded prints and logs (publishing)

## Changes
- set stdout/stderr of child process to DEVNULL on application launch if `sys.stdout` is `None`
- Foundry apps on windows
    - they are not launched using CMD.exe but using `CREATE_NEW_CONSOLE` flag
    - in their cases is stdout/stderr not redirected to DEVNULL because new console would not have output

## How to test
- all applications should work when launched using `openpype_gui` executable and from code
- change may affect mainly Foundry apps and Nonpython hosts
- test applications launch Launcher + Ftrack action
    - Nuke
    - Photoshop/Aftereffects
    - Tvpaint
    - Harmony
    - Blender
    - Maya

Depends on https://github.com/pypeclub/avalon-core/pull/354
